src/HotChocolate/ElasticSearch/src/Data.ElasticSearch.Driver/Driver/BoolOperation.cs

```diff
@@ -6,33 +6,33 @@ namespace HotChocolate.Data.ElasticSearch;
 public class BoolOperation : ISearchOperation
 {
     public BoolOperation(
-        IReadOnlyList<ISearchOperation> must,
-        IReadOnlyList<ISearchOperation> should,
-        IReadOnlyList<ISearchOperation> mustNot,
-        IReadOnlyList<ISearchOperation> filter)
+        IEnumerable<ISearchOperation> must,
+        IEnumerable<ISearchOperation> should,
+        IEnumerable<ISearchOperation> mustNot,
+        IEnumerable<ISearchOperation> filter)
     {
         Must = must;
         Should = should;
         MustNot = mustNot;
         Filter = filter;
     }
 
-    public IReadOnlyList<ISearchOperation> Must { get; }
+    public IEnumerable<ISearchOperation> Must { get; }
```

Comment:
Hmm why no IEnumerable and not IReadOnlyList?

Reply:
It is an effect by remvoing the tolist call.